src/WebJobs.Script/Diagnostics/FileLogger.cs

```diff
@@ -92,14 +92,8 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             try
             {
                 _fileWriter.AppendLine(formattedMessage);
-
-                // flush errors immediately
```

Comment:
Updated the PR to just remove this flush on error behavior. The underlying file is flushed on a 1s interval already, so we really won't save much time attempting to force a flush. Also, implementing the logic to force flush with a limited frequency would require synchronization and I don't want to put that on this logging hotpath.